distributed_index_tests/lib/user.py

```diff
@@ -10,102 +10,96 @@
 import re
 from collections import defaultdict
 
+
 class User:
-    def __init__(self, target_url, db, name, password, channels, uuid=False):
-        self.target_url = target_url
+    def __init__(self, target, db, name, password, channels, ):
+
         self.name = name
         self.password = password
         self.db = db
-        self.docs = {}
-        self.use_uuid_doc_names = uuid
+        self.docs_info = {}
         self.channels = list(channels)
-        _auth = base64.b64encode("{0}:{1}".format(self.name, self.password).encode())
-        self._auth = _auth.decode("UTF-8")
-        self.headers = {'Content-Type': 'application/json', "Authorization": "Basic {}".format(self._auth)}
-
 
-    def get_changes(self, with_docs=False):
-        headers = {
-            "Authorization": "Basic {}".format(self._auth)
-        }
+        self._url = target.url
+        auth = base64.b64encode("{0}:{1}".format(self.name, self.password).encode())
+        self._auth = auth.decode("UTF-8")
+        self._headers = {'Content-Type': 'application/json', "Authorization": "Basic {}".format(self._auth)}
 
-        if with_docs:
-            r = requests.get("{}/{}/_changes?include_docs=true".format(self.target_url, self.db), headers=headers)
-        else:
-            r = requests.get("{}/{}/_changes".format(self.target_url, self.db), headers=headers)
-        return json.loads(r.text)
+    def __str__(self):
+        return "USER: name={0} password={1} db={2} channels={3}".format(self.name, self.password, self.db, self.channels)
 
+    def set_target(self, target):
+        self._url = target.url
```

Comment:
We can directly assign the param to the instance variable. Not sure the use of creating a method to set it.